server/src/test/java/org/apache/druid/segment/indexing/granularity/IntervalsByGranularityTest.java

```diff
@@ -76,6 +79,21 @@ public void testDups()
     Assert.assertTrue(count == 61);
   }
 
+
+  @Test
+  public void testCondenseDoesNotMaterialize()
```

Comment:
It only does indirectly, if it materialized it would create 1 year worth of seconds amount of intervals (over 31 million). If it materialized I reasoned that it would slow down the test or something. I can remove it or rename it. Let me know if you think of a better way to test that. I don't want to measure memory before and after using the `Runtime.getRuntime()` methods because they are not very precise either.

Reply:
I changed the name of the method and added a comment inside the method to indicate intention..